### ISSUE TITLE
Traversal SDK - Is it possible to invoke a custom target via ProjectReference

### ISSUE THREAD

**Issue by ViktorHofer** [microsoft/MSBuildSdks#214]
Probably more an msbuild question but is it possible to use the Traversal SDK but not invoke the default target but a custom one?

**Comment by jeffkl** [comment]
If you want to just change the default target, you can set `<Project DefaultTargets="Foo" />` in the traversal project (https://docs.microsoft.com/en-us/visualstudio/msbuild/target-build-order?view=vs-2019#default-targets).  If you want to completely override the `Build` target, you can set the `CustomAfterNoTargets` property to a file to import after the `Build` target has been declared and you can declare your own.

**Comment by ViktorHofer** [comment]
Sorry, let me rephrase. Is it possible to invoke a custom target in the project referenced projects. I think I found the answer myself: https://github.com/dotnet/arcade/blob/5537c88d462e239a42375e19cc0ec2ab1ebeec64/src/Microsoft.DotNet.GenFacades/build/Microsoft.DotNet.GenFacadesNotSupported.targets#L22

**Comment by jeffkl** [comment]
Ah sorry yes, you can just set the `Targets` metadata on the `ProjectReference`.  Sorry I misunderstood!

**Comment by ViktorHofer** [comment]
Thanks. I believe that property is unfortunately not respected when building inside Visual Studio but shouldn't be relevant for this discussion. Closing.

**Comment by ViktorHofer** [comment]
cc @jkoritzinsky just told me that the Traversal SDK doesn't respect the `Targets` attribute.

**Comment by jkoritzinsky** [comment]
Here's the test case that I tried:

root.proj:
```xml
<Project Sdk="Microsoft.Build.Traversal/2.2.3">

   <ItemGroup>
      <ProjectReference Include="subproject.proj" Targets="Foo"/>
   </ItemGroup>
</Project>
```

subproject.proj:
```xml
<Project DefaultTargets="Build">
    <Target Name="Foo">
        <Message Importance="High" Text="Foo" />
    </Target>
    <Target Name="Build">
        <Message Importance="High" Text="Bar" />
    </Target>
</Project>
```

Command and output:
```
D:\scratch\MSBuild> dotnet msbuild .\root.proj
Microsoft (R) Build Engine version 16.8.0-preview-20475-05+aed5e7ed0 for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

  Bar
```

**Comment by jeffkl** [comment]
Thanks for reporting this, I'll get a fix out today: https://github.com/microsoft/MSBuildSdks/pull/216
